gon/point.py

```diff
@@ -0,0 +1,64 @@
+import math
+from typing import Tuple
+from weakref import WeakKeyDictionary
+
+from memoir import cached
+from reprit.base import generate_repr
+
+from .geometry import Geometry
+from .hints import Coordinate
+
+RawPoint = Tuple[Coordinate, Coordinate]
+
+
+class Point(Geometry):
+    __slots__ = '_x', '_y'
+
+    def __init__(self, x: Coordinate, y: Coordinate) -> None:
+        self._x = x
+        self._y = y
+
+    __repr__ = generate_repr(__init__)
+
+    def __hash__(self) -> int:
+        return hash((self._x, self._y))
+
+    def __eq__(self, other: 'Point') -> bool:
+        return (self._x == other._x and self._y == other._y
+                if isinstance(other, Point)
+                else NotImplemented)
```

Comment:
Again, in my opinion, it's better to return `False` instead of `NotImplemented` so it would be consistent with comparison of Python built-in types.

Reply:
it is consistent with built-in types, I've already said that, what `NotImplemented` is for

Reply:
Right, I was actually writing a counterexample with two classes that return `NotImplemented` but, to my surprise, comparison returns `False` there. Marking as resolved, and thanks! Today I learned something new :)